statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/StatefulFunctionsConfig.java

```diff
@@ -243,4 +246,34 @@ public void addAllGlobalConfigurations(Map<String, String> globalConfigurations)
   public void setGlobalConfiguration(String key, String value) {
     this.globalConfigurations.put(key, value);
   }
+
+  private static void validateStrictlyRequiredConfigs(Configuration configuration) {
```

Comment:
Can you extract the validation logic to a separate class?
It can be an inner static class for now, or a separate class with validation specific logic.